src/migrateStaticObservableMethodsRule.ts

```diff
@@ -0,0 +1,182 @@
+      const importStatementEnd = importStatement.getEnd();
+      ctx.addFailure(
+        importStatementStart,
+        importStatementEnd,
+        Rule.IMPORT_FAILURE_STRING,
+        Lint.Replacement.deleteFromTo(importStatementStart, importStatementEnd)
+      );
+    }
+  }
+}
+
+function isRxjsStaticOperator(node: ts.PropertyAccessExpression) {
+  return 'Observable' === node.expression.getText() && RXJS_OPERATORS.has(node.name.getText());
+}
+
+function isRxjsStaticOperatorCallExpression(node: ts.Node, typeChecker: ts.TypeChecker) {
+  // Expression is of the form fn()
+  if (!tsutils.isCallExpression(node)) {
+    return false;
+  }
+  // Expression is of the form foo.fn
+  if (!tsutils.isPropertyAccessExpression(node.expression)) {
+    return false;
+  }
+  // fn is one of RxJs instance operators
+  if (!isRxjsStaticOperator(node.expression)) {
+    return false;
+  }
+  // fn(): k. Checks if k is an observable. Required to distinguish between
+  // array operators with same name as RxJs operators.
+  if (!returnsObservable(node, typeChecker)) {
+    return false;
+  }
+  return true;
+}
+
+function findImportedRxjsOperators(sourceFile: ts.SourceFile): Set<string> {
+  return new Set<string>(
+    sourceFile.statements.filter(tsutils.isImportDeclaration).reduce((current, decl) => {
+      if (!decl.importClause) {
+        return current;
+      }
+      if (!decl.moduleSpecifier.getText().startsWith(`'rxjs'`)) {
+        return current;
+      }
+      if (!decl.importClause.namedBindings) {
+        return current;
+      }
+      const bindings = decl.importClause.namedBindings;
+      if (ts.isNamedImports(bindings)) {
+        return [
+          ...current,
+          ...(Array.from(bindings.elements) || []).map(element => {
+            return element.name.getText();
+          })
+        ];
+      }
+      return current;
+    }, [])
+  );
+}
+
+function operatorAlias(operator: string) {
+  return 'observable' + operator[0].toUpperCase() + operator.substring(1, operator.length);
+}
+
+function createImportReplacements(operatorsToAdd: Set<string>, startIndex: number): Lint.Replacement[] {
+  return [...Array.from(operatorsToAdd.values())].map(operator =>
+    Lint.Replacement.appendText(startIndex, `\nimport {${operator} as ${operatorAlias(operator)}} from 'rxjs';\n`)
+  );
+}
+
+/*
+ * https://github.com/ReactiveX/rxjs/tree/master/compat/add/observable
+ */
+const RXJS_OPERATORS = new Set([
+  'bindCallback',
+  'bindNodeCallback',
+  'combineLatest',
+  'concat',
+  'defer',
+  'empty',
+  'forkJoin',
+  'from',
+  'fromEvent',
+  'fromEventPattern',
+  'fromPromise',
+  'generate',
+  'if',
+  'interval',
+  'merge',
+  'never',
+  'of',
+  'onErrorResumeNext',
+  'pairs',
+  'rase',
+  'range',
+  'throw',
+  'timer',
+  'using',
+  'zip'
+]);
+
+// Not handling NEVER
+const OPERATOR_RENAMES: { [key: string]: string } = {
+  throw: 'throwError',
+  if: 'iif',
+  fromPromise: 'from'
```

Comment:
I remember this one caused some test failures in Google due to poorly written tests.
I vaguely remember the failing pattern was that suppose we have `fromPromise(foo())` in the source and `foo` is mocked the test. Now if `foo` is not properly mocked to return a promise RxJS v5.0 this doesn't throw an exception. After migrating to `from(foo())` in RxJS v6.0 this throws something like `You provided undefined where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.`

I think there's nothing much we need to do here. The users need to fix their tests in that case.
So this is just an FYI.